engine/src/test/scala/cromwell/SimpleWorkflowActorSpec.scala

```diff
@@ -1,80 +1,144 @@
+import com.typesafe.config.ConfigFactory
+import cromwell.SimpleWorkflowActorSpec._
+import cromwell.core.WorkflowId
 import cromwell.engine._
-import cromwell.engine.backend.WorkflowDescriptorBuilder
+import cromwell.engine.workflow.WorkflowActor
+import cromwell.engine.workflow.WorkflowActor.{StartNewWorkflow, StartWorkflowCommand}
+import cromwell.services.MetadataEvent
+import cromwell.services.MetadataServiceActor.PutMetadataAction
 import cromwell.util.SampleWdl
 import cromwell.util.SampleWdl.HelloWorld.Addressee
 
+import scala.concurrent.{Await, Promise}
 import scala.concurrent.duration._
 import scala.language.postfixOps
 
 
-class SimpleWorkflowActorSpec extends CromwellTestkitSpec with WorkflowDescriptorBuilder {
+object SimpleWorkflowActorSpec {
 
-  override implicit val actorSystem = system
+  object MetadataWatchActor {
+    def props(matcher: Option[FailureMatcher], promise: Promise[Unit]): Props = Props(MetadataWatchActor(matcher, promise))
+  }
 
-//  private def buildWorkflowFSMRef(sampleWdl: SampleWdl, rawInputsOverride: String):
-//  TestFSMRef[WorkflowState, WorkflowData, OldStyleWorkflowActor] = {
-//    val workflowSources = WorkflowSourceFiles(sampleWdl.wdlSource(), rawInputsOverride, "{}")
-//    val descriptor = materializeWorkflowDescriptorFromSources(workflowSources = workflowSources)
-//    TestFSMRef(new OldStyleWorkflowActor(descriptor))
-//  }
+  case class MetadataWatchActor(matcher: Option[FailureMatcher], promise: Promise[Unit]) extends Actor {
+    override def receive = matcher match {
+      case None =>
+        promise.trySuccess(())
+        Actor.emptyBehavior
+      case Some(m) => {
+        case PutMetadataAction(events) if m.matches(events) =>
+          promise.trySuccess(())
+          context.stop(self)
+      }
+    }
+  }
+
+  case class FailureMatcher(value: String) {
+    def matches(events: Traversable[MetadataEvent]): Boolean = {
+      events.exists(e => e.key.key == "failures[0]" && e.value.exists { v => v.value.contains(value) })
+    }
+  }
+
+  case class WorkflowActorandMetadataPromise(workflowActor: ActorRef, promise: Promise[Unit])
+}
+
+class SimpleWorkflowActorSpec extends CromwellTestkitSpec {
+
+  private def buildWorkflowActor(sampleWdl: SampleWdl, rawInputsOverride: String,
+                                 workflowId: WorkflowId = WorkflowId.randomId(),
+                                 matcher: Option[FailureMatcher] = None): WorkflowActorandMetadataPromise = {
+    val workflowSources = WorkflowSourceFiles(sampleWdl.wdlSource(), rawInputsOverride, "{}")
+    val promise = Promise[Unit]()
+    val watchActor = system.actorOf(MetadataWatchActor.props(matcher, promise), s"service-registry-$workflowId-${UUID.randomUUID()}")
+    val workflowActor = TestFSMRef(new WorkflowActor(workflowId, StartNewWorkflow, workflowSources, ConfigFactory.load(),
+      watchActor,
+      system.actorOf(Props.empty, s"workflow-copy-log-router-$workflowId-${UUID.randomUUID()}")),
```

Comment:
at that point you get into "what's a side effect", technically anything that throws should then have a `()`, or if one wanted to be super pedantic anything which used memory.

I don't care either way and only chimed in because @kshakir commented